wp-includes/js/src/site-icon-maskable/maskable-icon-controls.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Retrieves the translation of text.
+ *
+ * @see https://developer.wordpress.org/block-editor/packages/packages-i18n/
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * This module allows you to create and use standalone block editors. ;)
+ *
+ * @see https://developer.wordpress.org/block-editor/reference-guides/packages/packages-block-editor/
+ */
+import { InspectorControls } from '@wordpress/block-editor';
+
+/**
+ * This package includes a library of generic WordPress components
+ * to be used for creating common UI elements shared between screens
+ * and features of the WordPress dashboard.
+ *
+ * @see https://developer.wordpress.org/block-editor/reference-guides/packages/packages-components/
+ */
+import {
+	// __experimentalHStack as HStack,
+	Flex,
+	FlexBlock,
+	FlexItem,
+	PanelBody,
+	ToggleControl,
+} from '@wordpress/components';
+
+/**
+ * Core Data is a data module intended to
+ * simplify access to and manipulation
+ * of core WordPress entities.
+ *
+ * @see https://developer.wordpress.org/block-editor/reference-guides/packages/packages-core-data/
+ */
+import { store as coreStore, useEntityProp } from '@wordpress/core-data';
+
+/**
+ * WordPress’ data module serves as a hub
+ * to manage application state
+ * for both plugins and WordPress itself.
+ *
+ * @see https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/
+ */
+import { useSelect } from '@wordpress/data';
+
+export default function MaskableControls() {
+	const [siteIconMaskable, setSiteIconMaskable] = useEntityProp(
+		'root',
+		'site',
+		'site_icon_maskable'
+	);
+
+	// mainly borrowed from ...
+	const { isRequestingSiteIcon, siteIconUrl } = useSelect((select) => {
+		const { getEntityRecord, isResolving } = select(coreStore);
+		const siteData =
+			getEntityRecord('root', '__unstableBase', undefined) || {};
+
+		return {
+			isRequestingSiteIcon: isResolving('getEntityRecord', [
+				'root',
+				'__unstableBase',
+				undefined,
+			]),
+			siteIconUrl: siteData.site_icon_url,
+		};
+	}, []);
+
+	if (isRequestingSiteIcon) {
+		return null;
+	}
+
+	const siteIconStyle = {
+		clipPath: siteIconMaskable ? 'inset(10% round 50%)' : '',
+		width: '64px',
+	};
+
+	let siteIcon = <div style={siteIconStyle} />;
+
+	if (siteIconUrl) {
+		siteIcon = (
+			<img
+				alt={__('Site Icon')}
+				className="components-site-icon"
+				src={siteIconUrl}
+				width={64}
+				height={64}
+				style={siteIconStyle}
+			/>
+		);
+	}
```

Comment:
I tried selecting a non-square image as the Site Logo and this is resulting in an unexpected result for the maskable icon:

> ![image](https://user-images.githubusercontent.com/134745/222025613-b5c6d1c4-e1c0-41e5-b716-6a59e1e79532.png)

Three things aren't quite right here:

1. The toggle for a maskable icon is showing even if I've not selected that I want to use the Site Logo as the Site Icon. I should only see the toggle if I'm using the Site Logo as the Site Icon.
2. A Site Logo need not be square, but a Site Icon must be. Nevertheless, WP allows selection of a rectangular icon but then it crops it to be square. The same logic used for cropping to be square should also be replicated here in the preview. (Aside: I'm surprised that core doesn't have such a Site Icon preview when the "Use as site icon" toggle is checked, as there should be a way to see whether the square crop will even work. This would be an improvement for core.)
3. If I _change_ the selected image for the Site Logo while the block controls are open, the preview for the maskable icon does not update to use the newly selected image. See below:

> ![image](https://user-images.githubusercontent.com/134745/222026987-15124ffa-fbc2-46eb-9832-84893e33d62c.png)

Reply:
OMG. But you are totally right on all of your points. 

I'll try to find some solutions.